nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -364,7 +364,7 @@ in
         # If the empty string is assigned to this option, the list of commands to start is reset, prior assignments of this option will have no effect.
         ExecStart = [ "" ''${cfg.package}/bin/caddy run ${runOptions} ${optionalString cfg.resume "--resume"}'' ];
         # Validating the configuration before applying it ensures we’ll get a proper error that will be reported when switching to the configuration
-        ExecReload = [ "" ''${cfg.package}/bin/caddy reload ${runOptions} --force'' ];
+        ExecReload = [ "" ] ++ lib.optional cfg.enableReload ''${cfg.package}/bin/caddy reload ${runOptions} --force'';
```

Comment:
```suggestion
        ExecReload = [ "" ] ++ lib.optionals cfg.enableReload [ ''${lib.getExe cfg.package} reload ${runOptions} --force'' ];
```

Reply:
Still don't think we should use `lib.optionals` if it's just a single string (but `lib.optional`), but happy to do the `lib.getExe` suggestion.

Reply:
Technically both solutions are valid and this is mostly an opinionated change.

I see that many people are making mistakes in `nixpkgs`, mixing the use of `lib.optional` and `lib.optionals`. For all these reasons, I think using `lib.optionals` when dealing with an attributes which is a list makes more sense.

Also, using `lib.optionals` reduce the cognitive load for newcomers since the brackets needs to be explicitly added.

In any case, this won't prevent me from sleeping if you don't do the change, it's just based on my thin experience in the Nix world.

Reply:
Hmmh, I misinterpreted as a "request changes" type of review, because the old red review didn't get cleared.

I personally prefer using `lib.optional`. These are kinds of personal preferences, I usually leave it up to the authors of PRs.

Reply:
Updated.